ts/WoltLabSuite/Core/Component/User/List.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Object-based user list.
+ *
+ * @author  Marcel Werk
+ * @copyright  2001-2022 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Component/User/List
+ */
+
+import { dboAction } from "../../Ajax";
+import WoltlabCoreDialogElement from "../../Element/woltlab-core-dialog";
+import UiPagination from "../../Ui/Pagination";
+import { dialogFactory } from "../Dialog";
+
+export class UserList {
+  readonly #options: AjaxRequestOptions;
+  readonly #dialogTitle: string;
+  readonly #cache = new Map<number, string>();
```

Comment:
I've never been a fan of these caches in the frontend. Are they actually worth it in this case where the dialog is only opened after explicit interaction?